tests/test_gradient_boosting.py

```diff
@@ -166,6 +166,32 @@ def test_early_stopping_classification(data, scoring, validation_split, tol):
         assert gb.n_iter_ == max_iter
 
 
+def test_should_stop():
+
+    def should_stop(scores, n_iter_no_change, tol):
+        gbdt = GradientBoostingClassifier(n_iter_no_change=n_iter_no_change,
+                                          tol=tol)
+        return gbdt._should_stop(scores)
+
+    # not enough iterations
+    assert not should_stop([], n_iter_no_change=1, tol=0.001)
+
+    assert not should_stop([1, 1, 1], n_iter_no_change=5, tol=0.001)
+    assert not should_stop([1] * 5, n_iter_no_change=5, tol=0.001)
+
+    # still making significant progress up to tol
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.001)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=0.999)
+    assert not should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5,
+                           tol=5 - 1e-5)
```

Comment:
Do you mean `1 - 1e-5`?

Reply:
Hum you are right, it's me that was wrong with my 0.999 tol...

Reply:
Also I changed 

```
    assert _should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=2)
```

to

```
    assert _should_stop([1, 2, 3, 4, 5, 6], n_iter_no_change=5, tol=5)
```